### ISSUE TITLE
WrongEqualsTypeParameter does not ignore multi-parameter equals methods

### ISSUE THREAD

**Issue by FatihYalmanbas** [arturbosch/detekt#2230]
WrongEqualsTypeParameter

## Expected Behavior of the rule

As stated with the documentation provided, WrongEqualsTypeParameter should only check for equals methods with one parameter. 

If function is not top level and name is equals and has one parameter that is not Any? or Kotlin.Any? rule should fail.

## Context

For mocking purposes I delegate some of standard library methods to helper classes. such as 

```Kotlin
interface StringHelper {
    @Contract("null -> true")
    fun isEmpty(input: String?): Boolean

    @JvmDefault
    @Contract("null, null -> true")
    fun equals(left: String?, right: String?, ignoreCase: Boolean = false): Boolean
}

class StringHelperImpl : StringHelper {

    override fun isEmpty(input: String?) = input.isNullOrEmpty()

    override fun equals(left: String?, right: String?, ignoreCase: Boolean): Boolean = left.equals(right, ignoreCase)

}
```

In this particular case rule fails. I believe rule should ignore equals methods with more than one parameter.
